### DISCUSSION TITLE
How can I stop the CMake setup and autocoders running every time I rebuild?

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1676] [Q&A]
Hi,

When I am working on unit tests, it is common for me to fix a problem, rebuild and run just my current set of tests to make sure I've done the right thing, over and over again.

Whenever I rebuild, however, it looks like cmake is itself re-running `cmake ..` from the build cache, and all the autocoders are running, even though I haven't changed anything which would affect generated code, and even, I think, all the other necessary sources are being rebuilt. I may only need to build one source file and relink. I can't see that it should take more than a couple of seconds, but it actually takes 3.5 minutes.

Surely there is a way to prevent this?

**Comment by ThibFrgsGmz** [comment]
It seems that your problem is related to issue https://github.com/nasa/fprime/issues/1431. No reason has yet been found behind this.

**Reply by LeStarch** [reply]
I think @JackNWhite provided insight here (see discussion on issue #1431).   In short:

1. XML files are generated products now with FPP
2. XML files are still listed as influencing the CMake dependency graph

Thus we'll get a ping-pong in the generation.

> Note: even after a fix, changing .fpp models and CMakeList.txt files will still result in a regeneration.
